sql/scanner.go

```diff
@@ -0,0 +1,121 @@
+package sql
+
+import "fmt"
+
+type StringSlice struct {
+	Data []string
+}
+
+func (s *StringSlice) Scan(src any) error {
+	arr, ok := src.([]interface{})
+	if !ok {
+		return fmt.Errorf("unsupported value")
+	}
+
+	if len(arr) == 0 {
+		return nil
+	}
+
+	for _, elem := range arr {
+		str, ok := elem.(string)
+		if !ok {
+			return fmt.Errorf("unsupported value in array")
+		}
+
+		s.Data = append(s.Data, str)
+	}
+
+	return nil
+}
+
+type IntSlice struct {
+	Data []int
+}
+
+func (s *IntSlice) Scan(src any) error {
+	arr, ok := src.([]interface{})
+	if !ok {
+		return fmt.Errorf("unsupported value")
+	}
+
+	if len(arr) == 0 {
+		return nil
+	}
+
+	for _, elem := range arr {
+		switch num := elem.(type) {
```

Comment:
Which is the minimum Golang version library should support? Asking because depending on this we can use generics from Golang.

Reply:
Good observation!

Should probably work, version-wise. Especially since there won't be any old projects wanting to utilize this library, we could probably rely on Go 1.19+

(Go 1.18 was defined previously, which should also have generics, if I remember correctly.)

Reply:
Hey @ffrizzo @EtienneBruines - good catch! I have added generics already, so we can safely use them IMO.